packages/good-design/src/apps/bridge/MicroBridge.tsx

```diff
@@ -0,0 +1,232 @@
+import React, { useEffect, useCallback, useState } from "react";
+import {
+  Box,
+  Button,
+  Checkbox,
+  CheckIcon,
+  CloseIcon,
+  Flex,
+  FormControl,
+  HStack,
+  IconButton,
+  InfoOutlineIcon,
+  Popover,
+  Spinner,
+  Stack,
+  Switch,
+  Text,
+  WarningOutlineIcon
+} from "native-base";
+import { TransactionStatus } from "@usedapp/core";
+import { TokenInput } from "../../core";
+import { AddressInput, isAddressValid } from "../../core/inputs/AddressInput";
+import { BigNumber } from "ethers";
+import { ExplorerLink } from "../../core/web3/ExplorerLink";
+
+type OnBridge = (amount: string, sourceChain: string, target?: string) => Promise<void>;
+
+const Status = ({ result, ...props }: { result?: string }) => {
+  switch (result) {
+    case undefined:
+    case "Mining":
+    case "PendingSignature":
+    default:
+      return <Spinner {...props} />;
+    case "Success":
+      return <CheckIcon size="5" mt="0.5" color="emerald.500" {...props} />;
+    case "Fail":
+    case "Exception":
+      return <CloseIcon size="5" mt="0.5" color="danger.500" {...props} />;
+  }
+};
+
+const StatusBox = ({
+  txStatus,
+  text,
+  infoText,
+  sourceChain
+}: {
+  txStatus?: Partial<TransactionStatus>;
+  text: string;
+  infoText?: string;
+  sourceChain: "fuse" | "celo";
+}) => (
+  <Stack mt="2" alignItems="center" direction={["column", "column", "row"]}>
+    <HStack alignItems="center" flex="2 0">
+      <Box>
+        <Status result={txStatus?.status} />
+      </Box>
+      <Box flex={"1 1"}>
+        <Text color="emerald.500" fontSize="md" ml="2">
+          {text}
+        </Text>
+      </Box>
+      {infoText && (
+        <Popover
+          trigger={triggerProps => {
```

Comment:
no refs to StatusBox props - no reason to make it inline component
inline components are bad for optimization